src/test/groovy/org/dormitory/autobotsoccub/engine/GameEngineShould.groovy

```diff
@@ -0,0 +1,115 @@
+package org.dormitory.autobotsoccub.engine
+
+import org.dormitory.autobotsoccub.engine.exception.GameDoesNotExistException
+import org.dormitory.autobotsoccub.engine.model.Game
+import org.dormitory.autobotsoccub.engine.model.GameData
+import spock.lang.Specification
+
+import static org.dormitory.autobotsoccub.engine.model.PlayPosition.GK
+
+class GameEngineShould extends Specification {
+
+    def GK_A = 0
+    def FW_A = 1
+    def GK_B = 2
+    def FW_B = 3
+
+    Game game
+    GameEngine sut = new GameEngine()
+
+    def setup() {
+        game = Game.builder()
+                .goalKeeperA(GK_A)
+                .forwardA(FW_A)
+                .goalKeeperB(GK_B)
+                .forwardB(FW_B)
+                .build()
+    }
+
+    def "correctly started game"() {
+        when:
+        def newGame = sut.startGame(FW_A, game)
+        def existGame = sut.startGame(FW_A, game)
+
+        then:
+        newGame
+        !existGame
+    }
+
+    def "correctly executed game commands"() {
+        given:
+        sut.startGame(FW_A, game)
+        sut.metaClass.getCurrentGame << { Integer userId ->
```

Comment:
can this be moved to `setupSpec()` method?

Reply:
for what? We call this method only within the method

Reply:
yep, but the test is incorrect itself. in my prev comment (prev commit) I mentioned that this test should be split into smaller parts; again, this is _unit-test_
>>
1. "correctly count scored goal" (use dataTable to supply different scores (per user) that should be verified)
2. "fail to score" (think about edge cases)
**1 and 2 should test only ** `score` **method**

add corresponding tests for `autoScore` and `revert`